stats.php

```diff
@@ -109,6 +117,46 @@ function track_file_delete() {
 	] );
 }
 
+function track_xml_rpc_password_type( $xmlrpc_method ) {
+	// Skip tracking for non-XML-RPC requests.
+	if ( ! defined( 'XMLRPC_REQUEST' ) || ! XMLRPC_REQUEST ) {
+		return;
+	}
+
+	// Skip tracking for unauthenticated requests.
+	if ( ! is_user_logged_in() ) {
+		return;
+	}
+
+	// Skip tracking for Jetpack requests.
+	if ( vip_is_jetpack_request() ) {
+		return;
+	}
+
+	XML_RPC_Auth_Tracker::track( $xmlrpc_method );
+}
+
+function application_password_did_authenticate( $user ) {
+	// Only proceed if it's an XML-RPC request
+	if ( ! ( defined( 'XMLRPC_REQUEST' ) && XMLRPC_REQUEST ) ) {
+		return $user;
```

Comment:
```suggestion
		return;
```
@brunobasto the `application_password_did_authenticate` is an [action hook](https://developer.wordpress.org/reference/hooks/application_password_did_authenticate/) and not a filter. Therefore the `return $user` is not used anywhere, we can just `return`